Test proxy

nicharesuk opened the pull request:
Ok so I pushed a branch called "test_proxy" it now contains a make file for testing without having to use the build folder, all you have to do is make sure you have "concurrently" installed which is an npm package
and then run make, which will take you to localhost:3000 which is the react app, but all url calls from there will also be proxied to the flask app running on port 5000
to test this you can run :

curl http://127.0.0.1:3000/api/anime/300

it will show you that the proxying is working, and this is only for testing though, in production we will just be hosting the static files from the build folder

please let me know if anything is not working for you and I will help as soon as I can

also feel free to change the "300" to any number, it's just to test flask reading in parameters from the url

swindiggie commented:
Is the makefile necessary? We are already running scripts from the package.json, so I feel like we could just keep everything there for the sake of consistency.

For example, to run concurrently using `npm start` you can define this in the package.json:
```
"scripts": {
    "start": "concurrently \"node scripts/start.js\" \"python3 main.py\"",
    "build": "node scripts/build.js",
    "test": "node scripts/test.js --env=jsdom"
}
```

And according to this article, we don't have to install concurrently globally in order to run it if we use it from "scripts".
https://firstdoit.com/no-need-for-globals-using-npm-dependencies-in-npm-scripts-3dfb478908

nicharesuk commented:
You are right I will update that to get rid of the make file, I totally forgot about using npm scripts

nicharesuk committed:
Setting up testing environment files, adding infrastructure


Fixing 404 on default path


Remove make file in favor of npm scripts


Merge remote-tracking branch 'origin/test_proxy' into test_proxy


Revert "Merge remote-tracking branch 'origin/test_proxy' into test_proxy"

This reverts commit 73badb4c2666408c1adc1b634055ed1a6a170b0f.
